core/hash_map.h

```diff
@@ -74,13 +78,14 @@ class HashMapHahserDefault {
  * @param TKey  Key, search is based on it, needs to be hasheable. It is unique in this container.
  * @param TData Data, data associated with the key
  * @param Hasher Hasher object, needs to provide a valid static hash function for TKey
+ * @param Comparator comparator object, needs to be able to safely compare two TKey values
```

Comment:
And mind you, Go devs could have chosen to to put all NaNs into the same bin as well. It only requires them to make changes in the hashing function I linked to, and still won't require users to implement custom methods like Comparator does.

The nature of the workaround  is perpendicular whether you expect users to actively implement a certain function with certain behavior to make it work.